addons/l10n_in/models/account_invoice.py

```diff
@@ -101,7 +101,7 @@ def _post(self, soft=True):
         posted = super()._post(soft)
         gst_treatment_name_mapping = {k: v for k, v in
                              self._fields['l10n_in_gst_treatment']._description_selection(self.env)}
-        for move in posted.filtered(lambda m: m.country_code == 'IN'):
+        for move in posted.filtered(lambda m: m.company_id.account_fiscal_country_id.code == 'IN' and m.move_type in ['out_invoice', 'out_refund']):
```

Comment:
```suggestion
        for move in posted.filtered(lambda m: m.country_code == 'IN' and m.move_type in ['out_invoice', 'out_refund']):
```
Unnecessary change. 

It is weird because there is code below that checks if the journal is purchase.

Reply:
It's better to use `m.is_sale_document()` instead of `m.move_type in ['out_invoice', 'out_refund']` :)